src/identities.js

```diff
@@ -64,15 +70,47 @@ class Identities {
   }
 
   async verifyIdentity (identity) {
-    const verified = await this._keystore.verify(
+    if (!Identity.isIdentity(identity)) {
+      return false
+    }
+
+    const knownID = this._knownIdentities.get(identity.signatures.id)
+    if (knownID) {
+      return identity.id === knownID.id &&
+             identity.publicKey === knownID.publicKey &&
+             identity.signatures.id === knownID.signatures.id &&
+             identity.signatures.publicKey === knownID.signatures.publicKey
+    }
+
+    const verifyIdSig = await Keystore.verify(
       identity.signatures.id,
       identity.publicKey,
       identity.id
     )
-    return verified && Identities.verifyIdentity(identity)
+    if (!verifyIdSig) return false
+
+    const IdentityProvider = getHandlerFor(identity.type)
+    const verified = await IdentityProvider.verifyIdentity(identity)
+    if (verified) {
+      this._knownIdentities.set(identity.signatures.id, Identity.toJSON(identity))
+    }
+
+    return verified
   }
 
   static async verifyIdentity (identity) {
+    if (!Identity.isIdentity(identity)) {
+      return false
+    }
+
+    const verifyIdSig = await Keystore.verify(
```

Comment:
Which `Keystore` instance (or data directory) does this use?

Reply:
@haadcode it uses the Keystore's static `verify` method, it doesn't use a directory  https://github.com/orbitdb/orbit-db-keystore/blob/409d6d46f1260c2a00ae2920a068460a2298481c/src/keystore.js#L182

Reply:
Thanks @shamb0t, makes sense 👍

Reply:
@haadcode hmm after running tests with orbit-db its true that if a custom keystore is used then the custom keystore should also be used for verifying. Does it make sense that the `static verifyIdentity` would only verify identities which used `orbit-db-keystore`? (ie `this._keystore.verify` in the non-static method and `Keystore.verify` in the static method)

Reply:
You're right, perhaps we don't need the static method. I'd keep it in though in this PR and remove it later on. For now, we should be good, right?